runtime/filetype.lua

```diff
@@ -13,7 +13,13 @@ vim.api.nvim_create_autocmd({ 'BufRead', 'BufNewFile' }, {
   group = 'filetypedetect',
   callback = function(args)
     local ft, on_detect = vim.filetype.match({ buf = args.buf })
-    if ft then
+    if not ft then
+      -- Generic configuration file used as fallback
+      ft = require('vim.filetype.detect').conf(args.file, args.buf)
```

Comment:
Based on my comment below about `<amatch>` vs. `<afile>`, this should probably be
```suggestion
      ft = require('vim.filetype.detect').conf(args.match, args.buf)
```